### ISSUE TITLE
Early stopping: save model only if validation loss or perplexity improves

### ISSUE THREAD

**Issue by danielwatson6** [danielwatson6/hate-speech-project#28]
This would save time from re-training models until a specific number of steps (that must be computed manually since the number of training steps is determined by the number of epochs)
